lib/providers/analytics.dart

```diff
@@ -91,6 +93,31 @@ class Analytics {
     RudderController.instance.track('section_clicked', properties: getCommonData().putValue(map: event.toJson()));
   }
 
+  void searchPerformed(SearchPerformedEvent event) {
+    RudderController.instance.track('search_performed', properties: getCommonData().putValue(map: event.toJson()));
+  }
+
+  void searchResultClicked(BuildContext context) {
+    final searchAnalytics = InheritedData.read<SearchAnalytics>(context);
+    if (searchAnalytics == null) {
+      FirebaseCrashlytics.instance.recordError(Exception('Missing SearchAnalytics.'), StackTrace.current);
+    }
+    final searchItemAnalytics = InheritedData.read<SearchItemAnalytics>(context);
+    if (searchItemAnalytics == null) {
+      FirebaseCrashlytics.instance.recordError(Exception('Missing SearchItemAnalytics.'), StackTrace.current);
+    }
+
+    final event = SearchResultClickedEvent(
+      searchText: searchAnalytics!.searchText,
```

Comment:
This ! is very scary.
The problem here is that you're not returning after recordError(, so this code might run even searchAnalytics is null